{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "outputs": [],
   "source": [
    "from gpt_xml_helpers import *"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "outputs": [],
   "source": [
    "import io\n",
    "import os\n",
    "import time\n",
    "import pandas as pd\n",
    "import xml.etree.ElementTree as ET\n",
    "from typing import Dict\n",
    "from pathlib import Path"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "outputs": [],
   "source": [
    "# wrapper = GPTWrapper(\"gpt-3.5-turbo-1106\")\n",
    "wrapper = GPTWrapper(\"gpt-4-1106-preview\")"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "outputs": [],
   "source": [
    "data_path = Path('sample_xmls')\n",
    "xml_folder = data_path"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "outputs": [],
   "source": [
    "xml_files = []\n",
    "\n",
    "for root, dirs, files in os.walk(xml_folder):\n",
    "    for file in files:\n",
    "        if file.endswith('.xml'):\n",
    "            xml_files.append(os.path.join(root, file))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [],
   "source": [
    "def prompt_truncated_paper(wrapper, parsed_xml: Dict, max_tokens: int = 100000):\n",
    "    truncated_paper = truncate(\n",
    "        f\"\"\"Abstract:\n",
    "```\n",
    "{parsed_xml['abstract']}\n",
    "```\n",
    "\n",
    "Figures/Tables Captions:\n",
    "```\n",
    "{parsed_xml['figure_and_table_captions']}\n",
    "```\n",
    "\n",
    "Main Content:\n",
    "```\n",
    "{parsed_xml['main_content']}\n",
    "```\"\"\",\n",
    "        max_tokens,\n",
    "        wrapper,\n",
    "    )\n",
    "\n",
    "    text_to_send = f\"\"\"Your task now is to evaluate whether a scientific publication titled \"{parsed_xml['title']}\" includes information on how to obtain the data reported in the paper:\n",
    "\n",
    "{truncated_paper}\n",
    "\n",
    "\n",
    "======\n",
    "Your task:\n",
    "Determine whether this paper reports shared data.\n",
    "\n",
    "Start by \"Transparency outline:\".\n",
    "And then answer:\n",
    "\"1. List the types of data reported in the paper\"\n",
    "\"2. Species of organisms studied(e.g. human, mouse, rat, non-human primate, etc)\"\n",
    "\"3. How to obtain raw data\"\n",
    "\"4. Data sharing statement\"\n",
    "\"5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\"\n",
    "\n",
    "Start the csv with \"csv\" and finish it with \"/csv\".\n",
    "\n",
    "\"\"\"\n",
    "    return text_to_send"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [],
   "source": [
    "def step2_parse_xml(xml: str) -> Dict:\n",
    "    xml_file = io.StringIO(xml)\n",
    "    tree = ET.parse(xml_file)\n",
    "    root = tree.getroot()\n",
    "\n",
    "    title = get_article_title(root)\n",
    "    abstract = get_abstract(root)\n",
    "    introduction = get_section_text(root, section_title=\"Introduction\")\n",
    "    figure_and_table_captions = get_figure_and_table_captions(root)\n",
    "\n",
    "    # Get all section titles, including Figures\n",
    "    section_titles = [\n",
    "        sec.find(\"title\").text if sec.find(\"title\") is not None else \"\"\n",
    "        for sec in root.findall(\".//sec\")\n",
    "    ]\n",
    "\n",
    "    # Get Main_content section, including Introduction, but excluding Figures\n",
    "    main_content = get_main_content(root)\n",
    "\n",
    "    return {\n",
    "        \"title\": title,\n",
    "        \"abstract\": abstract,\n",
    "        \"introduction\": introduction,\n",
    "        \"figure_and_table_captions\": figure_and_table_captions,\n",
    "        \"section_titles\": section_titles,\n",
    "        \"main_content\": main_content,\n",
    "    }\n",
    "\n",
    "# adapt number of tokens here\n",
    "def step3_get_lm_review(wrapper, parsed_xml: Dict) -> Dict:\n",
    "    text_to_send = prompt_truncated_paper(wrapper, parsed_xml, 15000)\n",
    "    review_generated = wrapper.send_query(text_to_send, n_query=1)\n",
    "    return {\"text_to_send\": text_to_send, \"review_generated\": review_generated}"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [],
   "source": [
    "rtransparent_df = pd.read_csv(\"rtransparent_gpt_xmls.csv\")"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "outputs": [
    {
     "data": {
      "text/plain": "['PMC6467449',\n 'PMC6505707',\n 'PMC7132284',\n 'PMC7933130',\n 'PMC7938970',\n 'PMC8443060',\n 'PMC8795560',\n 'PMC9216730',\n 'PMC9279418',\n 'PMC9369370']"
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples = rtransparent_df['article'].tolist()\n",
    "samples"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "outputs": [],
   "source": [
    "xml_samples = [xml for xml in xml_files if os.path.basename(xml).strip('.xml') in samples]"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gpt-4-1106-preview\n",
      "# tokens sent to GPT: 11449\n",
      "PMC8795560\n",
      "**********************************\n",
      "rtransparent is open data: False\n",
      "rtransparent data statement: nan\n",
      "**********************************\n",
      "Transparency outline:\n",
      "\n",
      "1. List the types of data reported in the paper\n",
      "The paper is discussing policy and conceptual frameworks regarding genomic data but does not reference specific datasets or raw data generated by the research itself.\n",
      "\n",
      "2. Species of organisms studied(e.g. human, mouse, rat, non-human primate, etc)\n",
      "The paper does not detail specific organisms studied; it speaks broadly about global biodiversity.\n",
      "\n",
      "3. How to obtain raw data\n",
      "The paper does not offer raw data, nor does it present a method for obtaining such data. It notes there is \"no data underlying this work.\"\n",
      "\n",
      "4. Data sharing statement\n",
      "The paper states, \"Data Availability: There are no data underlying this work.\"\n",
      "\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "Since no data were reported and there are no datasets listed to obtain, this CSV will contain only the data sharing statement.\n",
      "\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"None\" | \"None\" | \"None\" | \"There are no data underlying this work\"\n",
      "/csv\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 12945\n",
      "PMC7938970\n",
      "**********************************\n",
      "rtransparent is open data: False\n",
      "rtransparent data statement: nan\n",
      "**********************************\n",
      "Transparency outline:\n",
      "\n",
      "1. List the types of data reported in the paper\n",
      "- Survey responses on leisure time physical activity (LTPA) from the NIH-AARP Diet and Health Study respondents\n",
      "- Healthcare cost data derived from Medicare claims data\n",
      "- Statistical modeling results including the association between LTPA trajectories and Medicare costs\n",
      "\n",
      "2. Species of organisms studied(e.g. human, mouse, rat, non-human primate, etc)\n",
      "- Human\n",
      "\n",
      "3. How to obtain raw data\n",
      "- The raw data sets used in the paper, specifically the NIH-AARP Diet and Health Study and Medicare claims data, were not provided in the text excerpt available. The typical process includes contacting the corresponding authors or institutions, checking any supplementary material, or using provided digital object identifiers (DOIs) that may link to publicly accessible datasets.\n",
      "\n",
      "4. Data sharing statement\n",
      "- A data-sharing statement is not explicitly mentioned in the provided excerpts.\n",
      "\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"Survey responses, Medicare claims data, Statistical modeling results\" | \"Human\" | \"\" | \"\"\n",
      "/csv\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 11399\n",
      "PMC7132284\n",
      "**********************************\n",
      "rtransparent is open data: True\n",
      "rtransparent data statement: the gobs hcp and pnc datasets have been described in detail in prior publications ( 30 32 34 56 58 31 59 si appendix methods i results iv the hcp data used in this study are available to investigators deemed to have a legitimate research use according to the restricted access data use terms described at https://www humanconnectome org https://www nimhgenetics org david glahn@childrens harvard edu john blangero@utrgv edu https://github com/aaronab local correlations of cortical thickness.\n",
      "**********************************\n",
      "Transparency outline:\n",
      "1. List the types of data reported in the paper\n",
      "- High-resolution brain MRI data\n",
      "- Cortical thickness maps\n",
      "- Genotype data\n",
      "\n",
      "2. Species of organisms studied(e.g. human, mouse, rat, non-human primate, etc)\n",
      "- Human\n",
      "\n",
      "3. How to obtain raw data\n",
      "- Human Connectome Project (HCP) data: available at https://www.humanconnectome.org, subject to Restricted Access Data Use Terms.\n",
      "- Philadelphia Neurodevelopmental Cohort (PNC) data: available through dbGaP (phs000607.v1.p1).\n",
      "- Genetics of Brain Structure and Function Study (GOBS) genotype data: available through dbGaP (phs001215.v2.p2).\n",
      "- GOBS imaging phenotypes: available from the National Institute of Mental Health Repository or by contacting the authors directly.\n",
      "\n",
      "4. Data sharing statement\n",
      "- Access to the HCP data used in this study is subject to ethical and scientific review, and investigators must have a legitimate research use according to Restricted Access Data Use Terms. \n",
      "- PNC data is accessible through dbGaP.\n",
      "- GOBS data is available through dbGaP for genotype data and from the National Institute of Mental Health Repository for imaging phenotypes. Potential users must apply for the data by contacting the authors, pass an ethical and scientific review, comply with all relevant guidelines, and complete a material transfer agreement.\n",
      "\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"High-resolution brain MRI data, Cortical thickness maps\" | \"Human\" | \"https://www.humanconnectome.org\" | \"Available to investigators deemed to have a legitimate research use according to the Restricted Access Data Use Terms.\"\n",
      "\"Genotype data\" | \"Human\" | \"dbGaP (phs001215.v2.p2)\" | \"Available through dbGaP subject to ethical and scientific review.\"\n",
      "\"Cortical thickness maps\" | \"Human\" | \"https://www.nimhgenetics.org\" | \"Available from the National Institute of Mental Health Repository or by contacting the authors for access to GOBS imaging phenotypes.\"\n",
      "\"/csv\"\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 11312\n",
      "PMC6505707\n",
      "**********************************\n",
      "rtransparent is open data: True\n",
      "rtransparent data statement: appendix a supplementary data the following are the supplementary data to this article: supplementary data 1 acknowledgements this research was supported by the intramural research program of the national center for complementary and integrative health nih appendix a supplementary data to this article can be found online at https://doi org/10 1016/j ynpai 2018 09 001\n",
      "**********************************\n",
      "Transparency outline:\n",
      "1. List the types of data reported in the paper:\n",
      "- Participant ratings of heat stimuli (pain ratings before and after placebo induction).\n",
      "- Sensory discrimination accuracy.\n",
      "- Effectiveness ratings of the analgesic cream.\n",
      "- Tables and figures with summarized results and statistical analyses.\n",
      "\n",
      "2. Species of organisms studied:\n",
      "- Human.\n",
      "\n",
      "3. How to obtain raw data:\n",
      "- The paper does not provide explicit information or instructions on how to obtain the raw data.\n",
      "\n",
      "4. Data sharing statement:\n",
      "- No such statement is provided within the provided sections of the paper.\n",
      "\n",
      "5. Print a pipe-separated csv:\n",
      "```\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"Pain ratings, Sensory discrimination accuracy, Effectiveness ratings\" | \"Human\" | \"\" | \"\"\n",
      "/csv\n",
      "```\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 12940\n",
      "PMC7933130\n",
      "**********************************\n",
      "rtransparent is open data: True\n",
      "rtransparent data statement: competing interests the authors declare no competing interest data availability em densities and protein models have been deposited in the electron microscopy data bank and protein data bank for the nact-citrate (emd-22457 7jsk) and nact-pf2 (emd-22456 7jsj) complexes.      data availability em densities and protein models have been deposited in the electron microscopy data bank and protein data bank for the nact-citrate (emd-22457\n",
      "**********************************\n",
      "Transparency outline:\n",
      "\n",
      "1. List the types of data reported in the paper:\n",
      "- Cryo-EM structural data\n",
      "- Cellular localization and proteolytic susceptibility analysis\n",
      "- Fluorescence size-exclusion chromatography (FSEC)\n",
      "- Size-exclusion chromatography (SEC)\n",
      "- Whole cell transport assay with radioactive citrate\n",
      "- Thermostabilization assay results\n",
      "- Tryptophan fluorescence quenching data\n",
      "- Multi-angle light scattering (MALS) data\n",
      "- Biochemical analysis of mutations responsible for SLC13A5-Epilepsy\n",
      "\n",
      "2. Species of organisms studied:\n",
      "- Human (HEK293 cells, SLC13A5 gene)\n",
      "\n",
      "3. How to obtain raw data:\n",
      "The paper mentions \"Source data are provided as a Source Data file.\" under the \"Protein purification\" section. However, the specific link or repository where the source data can be accessed is not explicitly stated in the text provided.\n",
      "\n",
      "4. Data sharing statement:\n",
      "No explicit data sharing policy statement is provided in the text excerpts given. To know the full details of the data sharing policy, one might need to refer to the journal's policy or the full version of the paper.\n",
      "\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\":\n",
      "```\n",
      "csv\n",
      "type_of_data | species | link_to_dataset | data_sharing_statement\n",
      "Cryo-EM structural data | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Cellular localization and proteolytic susceptibility analysis | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Fluorescence size-exclusion chromatography (FSEC) | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Size-exclusion chromatography (SEC) | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Whole cell transport assay with radioactive citrate | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Thermostabilization assay results | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Tryptophan fluorescence quenching data | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Multi-angle light scattering (MALS) data | Human | Not provided | Source data are provided as a Source Data file.\n",
      "Biochemical analysis of mutations responsible for SLC13A5-Epilepsy | Human | Not provided | Source data are provided as a Source Data file.\n",
      "/csv\n",
      "```\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 14045\n",
      "PMC8443060\n",
      "**********************************\n",
      "rtransparent is open data: False\n",
      "rtransparent data statement: nan\n",
      "**********************************\n",
      "Transparency outline:\n",
      "\n",
      "1. List the types of data reported in the paper\n",
      "- Colorectal cancer incidence and risk factors\n",
      "- Baseline demographic and lifestyle characteristics\n",
      "- Food consumption frequencies\n",
      "- Intervention groups and supplementation details\n",
      "- Statistical analysis outcomes including relative risks (RRs), hazard ratios, 95% confidence intervals (CIs), and P-values\n",
      "\n",
      "2. Species of organisms studied(e.g. human, mouse, rat, non-human primate, etc)\n",
      "- Human\n",
      "\n",
      "3. How to obtain raw data\n",
      "- The paper does not provide explicit instructions or a repository link for obtaining the raw data.\n",
      "\n",
      "4. Data sharing statement\n",
      "- The paper does not include a data sharing statement.\n",
      "\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "```\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"Colorectal cancer incidence and risk factors | Baseline demographic and lifestyle characteristics | Food consumption frequencies | Intervention groups and supplementation details | Statistical analysis outcomes\" | \"Human\" | \"\" | \"\"\n",
      "/csv\n",
      "```\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 10556\n",
      "PMC9369370\n",
      "**********************************\n",
      "rtransparent is open data: False\n",
      "rtransparent data statement: nan\n",
      "**********************************\n",
      "Transparency outline:\n",
      "1. List the types of data reported in the paper\n",
      "2. Species of organisms studied(e.g. human, mouse, rat, non-human primate, etc)\n",
      "3. How to obtain raw data\n",
      "4. Data sharing statement\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\n",
      "1. The types of data reported in the paper include:\n",
      "   - Fatty acid composition of diets\n",
      "   - Hepatic fatty acid composition\n",
      "   - Plasma and hepatic lipid profiles\n",
      "   - Aortic atherosclerosis quantification\n",
      "   - Plasma cytokine levels\n",
      "   - Gene expression in liver and white adipose tissue\n",
      "   - Biochemical analyses data (e.g., lipoprotein fractions)\n",
      "   - Atherosclerotic lesion and aortic calcification analyses\n",
      "\n",
      "2. Species of organisms studied: Mouse (specifically, low-density lipoprotein receptor knock-out (LDLR-KO) mice)\n",
      "\n",
      "3. How to obtain raw data: The paper does not provide explicit instructions or links on how to obtain the raw data. Normally, raw data can be requested from the corresponding author, whose contact information is typically provided in the published paper.\n",
      "\n",
      "4. Data sharing statement: The paper does not include a data sharing statement.\n",
      "\n",
      "5.\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"Fatty acid composition of diets, Hepatic fatty acid composition, Plasma and hepatic lipid profiles, Aortic atherosclerosis quantification, Plasma cytokine levels, Gene expression in liver and white adipose tissue, Biochemical analyses data, Atherosclerotic lesion and aortic calcification analyses\" | \"Mouse\" | \"\" | \"No data sharing statement provided\"\n",
      "/csv\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 8625\n",
      "PMC9279418\n",
      "**********************************\n",
      "rtransparent is open data: True\n",
      "rtransparent data statement: data availability cryo-em density maps and the atomic model of prp sc emd-25824\n",
      "**********************************\n",
      "Transparency outline:\n",
      "The scientific publication titled \"Cryo-EM structure of anchorless RML prion reveals variations in shared motifs between distinct strains\" appears to have shared relevant data in accordance with scientific transparency norms.\n",
      "\n",
      "1. List the types of data reported in the paper\n",
      "- Cryo-EM-based structural data of infectious brain-derived fibrils\n",
      "- PAGE gels and Western blot analysis data\n",
      "- Mass spectrometry analysis data\n",
      "- Infectivity bioassay results using transgenic mice\n",
      "- Cryo-electron tomography data and analysis\n",
      "- Helical reconstruction data obtained through image processing\n",
      "- Molecular dynamics simulation data\n",
      "\n",
      "2. Species of organisms studied\n",
      "- Mouse (Mus musculus)\n",
      "\n",
      "3. How to obtain raw data\n",
      "The paper itself does not explicitly state how to obtain the raw data, but it does mention methods and supplemental information that might contain details on how the data can be accessed. For precise instructions on obtaining the data, one typically would check the \"Supplementary Information\" section, the associated \"Reporting Summary,\" or directly contact the corresponding authors.\n",
      "\n",
      "4. Data sharing statement\n",
      "The paper does not include an explicit data sharing statement within the excerpt provided. However, most journals require authors to make data available upon reasonable request, and raw data can often be found in supplemental files or deposited in public repositories linked to the publication.\n",
      "\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "```plaintext\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"Cryo-EM-based structural, PAGE gels, Western blot, Mass spectrometry, Infectivity bioassay results, Cryo-electron tomography, Helical reconstruction, Molecular dynamics simulation\" | \"Mouse\" | \"Not specified in the provided content\" | \"Not explicitly stated in the provided content\"\n",
      "/csv\n",
      "```\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 2233\n",
      "PMC9216730\n",
      "**********************************\n",
      "rtransparent is open data: False\n",
      "rtransparent data statement: nan\n",
      "**********************************\n",
      "Transparency outline:\n",
      "\n",
      "1. List the types of data reported in the paper\n",
      "- Clinical data (patient characteristics, rebound symptoms)\n",
      "- Virologic data (viral load, PCR cycle thresholds, viral sequencing, virus isolation)\n",
      "- Immunologic data (antibody levels, T-cell counts and responses, soluble biomarkers, inflammatory markers)\n",
      "\n",
      "2. Species of organisms studied\n",
      "- Human\n",
      "\n",
      "3. How to obtain raw data\n",
      "- The paper suggests that detailed methods and additional data are provided in the supplementary materials (indicated as \"supplement\"). However, no explicit mention of how to access the raw data is provided in the excerpt. Typically, one can obtain such data by checking the supplementary materials or by contacting the authors directly, often provided in the supplementary material or the main text of the publication.\n",
      "\n",
      "4. Data sharing statement\n",
      "- No data sharing statement is explicitly provided in the abstract, main content, or figures/tables captions included in the excerpt.\n",
      "\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"Clinical, Virologic, Immunologic\" | \"Human\" | \"Not Provided\" | \"Not Provided\"\n",
      "/csv\n",
      "**********************************\n",
      "\n",
      "# tokens sent to GPT: 13474\n",
      "PMC6467449\n",
      "**********************************\n",
      "rtransparent is open data: True\n",
      "rtransparent data statement: file s1 table ghana breast health study recruiting population controls minimal dataset (xlsx\n",
      "**********************************\n",
      "Transparency outline:\n",
      "1. List the types of data reported in the paper\n",
      "- Demographic data of control participants\n",
      "- Questionnaire responses regarding medical history, reproductive history, body size, family history of breast cancer, and lifestyle factors\n",
      "- Measurements of weight, height, waist and hip circumference\n",
      "- Saliva and blood samples\n",
      "\n",
      "2. Species of organisms studied\n",
      "- Human\n",
      "\n",
      "3. How to obtain raw data\n",
      "- There is no clear indication in the provided content that raw data is shared or how to obtain it. To access the raw data, one would typically need to contact the authors or access a data repository if any is listed.\n",
      "\n",
      "4. Data sharing statement\n",
      "- No explicit data sharing statement is provided in the provided content.\n",
      "\n",
      "5. Print a pipe separated csv with: \"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\n",
      "csv\n",
      "\"type_of_data\" | \"species\" | \"link_to_dataset\" | \"data_sharing_statement\"\n",
      "\"Demographic data, Questionnaire responses, Body measurements, Biological samples\" | \"Human\" | \"\" | \"No data sharing statement provided\"\n",
      "/csv\n",
      "**********************************\n",
      "\n"
     ]
    }
   ],
   "source": [
    "full_output = \"\"\n",
    "full_output += wrapper.model_name\n",
    "print(full_output)\n",
    "for xml in xml_samples:\n",
    "    current_output = \"\"\n",
    "    article = os.path.basename(xml).strip('.xml')\n",
    "    with open(xml) as xml_f:\n",
    "        current_output += article\n",
    "        current_output += '\\n**********************************\\n'\n",
    "        current_output += f\"rtransparent is open data: {rtransparent_df[rtransparent_df['article'] == article]['is_open_data'].iloc[0]}\\n\"\n",
    "        current_output += f\"rtransparent data statement: {rtransparent_df[rtransparent_df['article'] == article]['open_data_statements'].iloc[0]}\\n\"\n",
    "        current_output += '**********************************\\n'\n",
    "        parsed_xml = step2_parse_xml(xml_f.read())\n",
    "        generated = step3_get_lm_review(wrapper, parsed_xml)\n",
    "        current_output += generated[\"review_generated\"]\n",
    "        current_output += '\\n**********************************\\n'\n",
    "        print(current_output)\n",
    "        time.sleep(60)\n",
    "    full_output += current_output"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}